projects/driver_controls/inc/stm32f0xx/ads1015.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+// Module for using ads1015
+// GPIO and Interrupt, I2C should be initiated
+// The user also needs to create a struct ADS1015Data which would persist accross functions
+// start the ads1015 using ads1015_init and then configure channels using ads1015_config_channel
+// read raw and converted values using ads1015_read_raw and ads1015_read_converted.
+#include <stdbool.h>
+#include "gpio.h"
+#include "i2c.h"
+#include "status.h"
+
+typedef enum {
+  ADS1015_ADDRESS_GND = 0,
+  ADS1015_ADDRESS_VDD,
+  ADS1015_ADDRESS_SDA,
+  ADS1015_ADDRESS_SCL,
+  NUM_ADS1015_ADDRESSES,
+} ADS1015Address;
+
+typedef enum {
+  ADS1015_CHANNEL_0 = 0,
+  ADS1015_CHANNEL_1,
+  ADS1015_CHANNEL_2,
+  ADS1015_CHANNEL_3,
+  NUM_ADS1015_CHANNELS,
+} ADS1015Channel;
+
+typedef void (*ADS1015Callback)(const GPIOAddress *address, void *context);
+
+typedef struct ADS1015Data {
+  I2CPort i2c_port;
+  ADS1015Address i2c_addr;
+  GPIOAddress *ready_pin;
+  uint8_t channel_readings[NUM_ADS1015_CHANNELS][2];
+  ADS1015Channel current_channel;
+  bool channel_enable[NUM_ADS1015_CHANNELS];
+  ADS1015Callback channel_callback[NUM_ADS1015_CHANNELS];
+  void *callback_context[NUM_ADS1015_CHANNELS];
+} ADS1015Data;
```

Comment:
If this is storage, most of our modules use `[Module]Storage` as a naming scheme. Also, based on discussion with @ckitagawa, I think we're going to standardize on avoiding multiple capitals in PascalCase names. See ELEC-334 for an example and some more information.